DeviceDetector.php

```diff
@@ -213,12 +213,16 @@ public function __call(string $methodName, array $arguments): bool
     {
         foreach (AbstractDeviceParser::getAvailableDeviceTypes() as $deviceName => $deviceType) {
             if (\strtolower($methodName) === 'is' . \strtolower(\str_replace(' ', '', $deviceName))) {
+                $this->parse();
```

Comment:
that would actually not be needed, as `getDevice` would already trigger a `parser`

Reply:
I don't think it's a good idea to run parser in every methods isBot isTablet etc.
we have a special method to check whether the parser method was started

https://github.com/matomo-org/device-detector/blob/master/DeviceDetector.php#L541-L555

I think it should just be a check, if the parser was not called, then a fatal exception
e.g.
```php
foreach (AbstractDeviceParser::getAvailableDeviceTypes() as $deviceName => $deviceType) {
            if (\strtolower($methodName) === 'is' . \strtolower(\str_replace(' ', '', $deviceName))) {
                if ( false === $this->isParse()) {
                      throw new \Exception("method parse() has not been called before");
                }
```
of course, if our goal is to force the user to call the parse() method